src/lib/y2packager/known_repositories.rb

```diff
@@ -71,6 +71,12 @@ def read_repositories
       end
 
       status = YAML.load_file(status_file)
+
+      if !status
+        log.info("Status file #{status_file} is empty or corrupted")
+        return []
+      end
```

Comment:
sadly, this is not enough, as Psych uses exceptions:

try yourself:
```
irb(main):002:0> YAML.load_file("/dev/zero")
Psych::SyntaxError ((/dev/zero): control characters are not allowed at line 1 column 1)
```
Your fix works only for empty or comment only `---` content.

Reply:
well, in this case status file is hard coded and it doesn't seem to be user edited. 

moreover my fix was intended for content processing - parser itself didn't crash (no exception from the parser but nil / FalseClass exception on the line bellow)

but ok, i'll improve slightly more.